README.md

```diff
@@ -1,67 +1,41 @@
+- (or) Create a file `~/.github` on Go Server and on all Go Agent machines, with the following contents.
 ```
 login=johndoe
 password=thisaintapassword
+```
+
+Github Enterprise:
+- If you intend to use this plugin with 'Github Enterprise' then add this in `~/.github` (which needs to be available on Go Server and on all Go Agent machines)
+```
 # for enterprise installations - Make sure to add /api/v3 to the hostname
 # ignore this field or have the value to https://api.github.com
 endpoint=http://code.yourcompany.com/api/v3
 ```
-- Lets assume you already have a pipeline "ProjectA" for one of your Github repos
-- Do a 'Extract Template' from the pipeline (if not present already)
-- Now clone your "ProjectA" pipeline to "ProjectA-PullRequests"
-- Since SCM notification end point feature still doesn't have UI support for adding custom SCM Materials, you need to make some changes in the config.xml
-- Go to "Admin" -> "Config XML"
-- Add the following line in your xml
-```
-<?xml version="1.0" encoding="utf-8"?>
-<cruise ....>
-  <scms>
-    <!-- You might have to manually edit the id part here for every new material -->
-    <scm id="34293182-33e4-4459-a686-276d70387dfb" name="test">
-      <pluginConfiguration id="github.pr" version="0.1" />
-      <configuration>
-        <property>
-          <key>url</key>
-          <value>git@github.com:user-name/project-a.git</value>
-        </property>
-      </configuration>
-    </scm>
-  </scms>
 
-  <piplines>
-    <pipeline name="Project-A-PullRequests" template="Project-A-Build">
-      <materials>
-        <!-- Make sure the ref="..." value matches the one specified above -->
-        <scm ref="34293182-33e4-4459-a686-276d70387dfb">
-        </scm>
-      </materials>
-    </pipeline>
-  </pipelines>
-</cruise>
-```
+## Behavior
+- First run of the new pipeline will be off of 'master' branch. This is prepare a setup base PR-Revision map. It also serves as sanity check for newly created pipeline.
+- From then on, any new change (new PR create / new commits to existing PR) will trigger the new pipeline. Only the top commit in the PR will show up in build cause.
 
 ## To Dos
 - Clean up the code esp. the JSON SerDe part
 - Add proper tests around the plugin
-- Add support for [Github's commit status](https://developer.github.com/v3/repos/statuses/) API to push build status to Github. May be a separate task plugin?
+- Add support for [Github's commit status](https://developer.github.com/v3/repos/statuses/) API to push build status to Github. May be a separate task/notification plugin?
 
 ## FAQs
 
-### How Pull Requests are Built?
-We periodically poll for new PRs on the given repository. We build a pull request when it's first opened and when commits are added to the pull request throughout its lifetime.
-Rather than test the commits from the branches the pull request is sent from, we test the merge between the origin and the upstream branch.
-
-### Pull Request isn't being built
```

Comment:
We never used to merge on our side, we take the PR/merge/ commit that Github creates. We still do only that, isn't? The `PR_MERGE_PREFIX` actually points to the merge commits on the PRs. Please help me understand.